client/client.go

```diff
@@ -0,0 +1,281 @@
+		version = VersionNine
+	} else if strings.HasPrefix(metadata.Config.Version, VersionTen) {
+		version = VersionTen
+	} else {
+		version = UnknownVersion
+	}
+	return OpenResultsWithVersion(filename, version)
+}
+
+// OpenResultsWithVersion opens the file and sets the version.
+// Generally it's ok to use the OpenResults function to autodetect the results version.
+func OpenResultsWithVersion(filename, version string) (*Results, error) {
+	// TODO(chuckha) ensure we support the passed in version
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	zr, err := gzip.NewReader(f)
+	if err != nil {
+		return nil, err
+	}
+	return &Results{zr, version}, nil
+}
+
+// tarFileInfo implements os.FileInfo and extends the Sys() method to
+// return a reader to a file in a tar archive.
+type tarFileInfo struct {
+	os.FileInfo
+	io.Reader
+}
+
+// Sys is going to be an io.Reader to a file in a tar archive.
+// This is how data is extracted from the results file.
+func (t *tarFileInfo) Sys() interface{} {
+	return t.Reader
+}
+
+// IterateFilesWithPrefix will go through every file in the results file and
+// run the walkfn on any file that matches the prefix passed in.
+func (r *Results) IterateFilesWithPrefix(prefix string, walkfn filepath.WalkFunc) error {
+	tr := tar.NewReader(r)
+	var err error
+	var header *tar.Header
+	for {
+		header, err = tr.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		if strings.HasPrefix(header.Name, "./"+prefix) {
+			info := &tarFileInfo{
+				header.FileInfo(),
+				tr,
+			}
+			// Trim the path to not expose the version differences of result files.
+			err = walkfn(strings.TrimPrefix(header.Name, "./"+prefix), info, err)
+		}
+	}
+	return nil
+}
+
+// NamespacedResources iterates and executes walkfn for every namespaced resource file.
+func (r *Results) NamespacedResources(walkfn filepath.WalkFunc) error {
+	return r.IterateFilesWithPrefix(NamespacedResourcesDir, walkfn)
+}
+
+// NonNamespacedResources iterates and executes walkfn for every non-namespaced resource file.
+func (r *Results) NonNamespacedResources(walkfn filepath.WalkFunc) error {
+	return r.IterateFilesWithPrefix(nonNamespacedResources(r.Version), walkfn)
+}
+
+// Hosts iterates and executes walkfn for all files relating to the host data.
+func (r *Results) Hosts(walkfn filepath.WalkFunc) error {
+	return r.IterateFilesWithPrefix(HostsDir, walkfn)
+}
+
+// Plugins walks each plugin file executing walkfn on each.
+func (r *Results) Plugins(walkfn filepath.WalkFunc) error {
+	return r.IterateFilesWithPrefix(PluginsDir, walkfn)
+}
+
+// PodLogs walks each pod log file executing walkfn on each.
+func (r *Results) PodLogs(walkfn filepath.WalkFunc) error {
+	return r.IterateFilesWithPrefix(PodLogs, walkfn)
+}
+
+// Metadata extracts the metadata that Sonobuoy gathers from the results.
+// This function does trial and error to detect the version of the results file.
+func (r *Results) Metadata() *Metadata {
```

Comment:
I tried to address that in the comments -- maybe a rewording of something like

"This does not depend on `r.Version` because this is the method used to discover the version of the results."